spyder/plugins/variableexplorer/widgets/namespacebrowser.py

```diff
@@ -46,9 +46,9 @@
 SUPPORTED_TYPES = get_supported_types()
 
 if PY2:
-    VALID_VARIABLE_CHARS = r"^[^\W]\w*\Z"
+    VALID_VARIABLE_CHARS = r"[^\W]\w*\Z"
 else:
-    VALID_VARIABLE_CHARS = r"^[^\W]\w*\Z"
+    VALID_VARIABLE_CHARS = r"[^\w+*=¡!¿?'\"#$%&()/<>\-\[\]{}^`´;,|¬]*\w"

```

Comment:
@dalthviz are you checking is a variable has a valid name?

If that is the case, something like https://stackoverflow.com/a/29586366 might be useful?

Reply:
I was thinking the validation per character, so in Python 2 something like numbers, underscore and letters (ASCII lower or uppercase) should be the only valid ones. In the case of Python 3, the set of valid characters increases to anything besides the ones for operations (and in general the characters in the regex), allowing to use the filter with something like this:

![image](https://user-images.githubusercontent.com/16781833/61755745-95aae400-ad7e-11e9-911c-9a32ab73a159.png)

Reply:
Understood :-)

Reply:
Yep, @dalthviz is right: we need to allow users to filter any kind of utf-8 character.